contracts/strategies/BenqiStrategyV1.sol

```diff
@@ -0,0 +1,357 @@
+        require(_swapPairToken0 > address(0), "Swap pair 0 is necessary but not supplied");
+        require(_swapPairToken1 > address(0), "Swap pair 1 is necessary but not supplied");
+
+        require(
+            address(rewardToken0) == IPair(address(_swapPairToken0)).token0() ||
+            address(rewardToken0) == IPair(address(_swapPairToken0)).token1(),
+            "Swap pair 0 does not match rewardToken0"
+        );
+
+        require(
+            address(rewardToken1) == IPair(address(_swapPairToken0)).token0() ||
+            address(rewardToken1) == IPair(address(_swapPairToken0)).token1(),
+            "Swap pair 0 does not match rewardToken1"
+        );
+
+        require(
+            address(depositToken) == IPair(address(_swapPairToken1)).token0() ||
+            address(depositToken) == IPair(address(_swapPairToken1)).token1(),
+            "Swap pair 1 does not match depositToken"
+        );
+
+        require(
+            address(rewardToken1) == IPair(address(_swapPairToken1)).token0() ||
+            address(rewardToken1) == IPair(address(_swapPairToken1)).token1(),
+            "Swap pair 1 does not match rewardToken1"
+        );
+
+        swapPairToken0 = IPair(_swapPairToken0);
+        swapPairToken1 = IPair(_swapPairToken1);
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(tokenDelegator), type(uint256).max);
+        tokenDelegator.approve(address(tokenDelegator), type(uint256).max);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "BenqiStrategyV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount), "BenqiStrategyV1::transfer failed");
+        _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(amount));
+        totalDeposits = totalDeposits.add(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        uint depositTokenAmount = getDepositTokensForShares(amount);
+        if (depositTokenAmount > 0) {
+            uint toWithdraw = _withdrawDepositTokens(depositTokenAmount);
```

Comment:
should this also pass `amount` (in shares)?

L173 multiplies underlying (e.g. DAI) and deposit token (e.g. DAI) and divides by total supply (e.g. YRT).

Reply:
Alternative, L173 could simply use `amount` since the math is already done in L159